### PR TITLE
Fix errors reported by telemetry

### PR DESCRIPTION
**Pull request by evidolob**
### What does this PR do?
It provided fixes for error reported by telemetry, like https://github.com/redhat-developer/vscode-yaml/issues/653 and https://github.com/redhat-developer/vscode-yaml/issues/651
Also it fixes `TypeError: Cannot read property 'offset' of undefined`,
`TypeError: e.indexOf is not a function at w` and `TypeError: Cannot read property 'type' of undefined`

### What issues does this PR fix or reference?
Fix: https://github.com/redhat-developer/vscode-yaml/issues/653
Fix: https://github.com/redhat-developer/vscode-yaml/issues/651

### Is it tested? How?
with tests

### CODE REVIEW

File: src/languageservice/parser/ast-converter.ts [hunk 1]
```diff
@@ -138,6 +142,18 @@ function convertScalar(node: Scalar, parent: ASTNode): ASTNode {
       result.isInteger = Number.isInteger(result.value);
       return result;
     }
+    default: {
+      if (node.tag === 'tag:yaml.org,2002:binary') {
```
Comment (gorkem) [inline src/languageservice/parser/ast-converter.ts:146]: Why do we need this `if` in here? It looks like we are constructing exactly the same result to return?

File: src/languageservice/parser/jsonParser07.ts [hunk 2]
```diff
@@ -233,6 +233,16 @@ export function asSchema(schema: JSONSchemaRef): JSONSchema {
   if (isBoolean(schema)) {
     return schema ? {} : { not: {} };
   }
+  if (schema === undefined) {
```
Comment (gorkem) [inline src/languageservice/parser/jsonParser07.ts:236]: Shouldn't this `if` come before the one above and simplfy the body of it?

File: src/languageservice/parser/jsonParser07.ts [hunk 3]
```diff
@@ -233,6 +233,16 @@ export function asSchema(schema: JSONSchemaRef): JSONSchema {
...
+    return schema;
+  }
+  if (typeof schema !== 'object') {
+    // we need to report this case as JSONSchemaRef MUST be an Object or Boolean
+    console.warn(`Wrong schema: ${JSON.stringify(schema)}, it MUST be an Object or Boolean`);
```
Comment (gorkem) [inline src/languageservice/parser/jsonParser07.ts:241]: What do we expect to happen when we log this ? Do we expect users to refer to the logs?

**Review by gorkem — CHANGES_REQUESTED** [review]

File: src/languageservice/parser/jsonParser07.ts (hunk 3 above)
Comment (evidolob) [reply src/languageservice/parser/jsonParser07.ts:241]: That is can happen when user use not entirely valid JSON Schema. 
I not sure how we should report that to user. As that check can be called multiple times, so I don't want to spam user with multiple error/warning notifications.

**Comment by evidolob** [comment]
@gorkem Could you look on this?

**Review by gorkem — APPROVED** [review]

**Review by msivasubramaniaan — APPROVED** [review]

**Pull request merged**